### PR TITLE
Add horizontal alignment to column configuration

### PR DESCRIPTION
**Pull request by mperrotti**
Closes https://github.com/github/primer/issues/1959

### Screenshots

Kapture 2023-03-23 at 10 25 54

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/DataTable/Table.tsx [hunk 1]
```diff
@@ -55,9 +56,22 @@ const StyledTable = styled.table<React.ComponentPropsWithoutRef<'table'>>`
...
   }
 
+  .TableHeader[data-cell-align='end'],
+  .TableCell[data-cell-align='end'] {
+    text-align: end;
+    display: flex;
+    justify-content: flex-end;
+  }
+
+  .TableHeader[data-cell-align='end'] > button {
```
Comment (joshblack) [inline src/DataTable/Table.tsx:70]: Could we add a class to the table sort button instead of using an element selector?

**Review by joshblack — APPROVED** [review]
Looks great! Just left a comment for the CSS Selector.

Also wanted to ask if we wanted a shared `Alignment` or `BlockAlignment` type after seeing the `ToggleSwitch` update. I think seeing `CellAlignment` there was unexpected since I didn't think of it as a cell but totally get wanting alignment values to be consistent across the library/design system 👍

File: src/DataTable/Table.tsx (hunk 1 above)
Comment (mperrotti) [reply src/DataTable/Table.tsx:70]: Sure!

**Review by joshblack — APPROVED** [review]
🥳

**Pull request merged**